src/server/rdb_load.cc

```diff
@@ -2787,6 +2787,10 @@ void RdbLoader::PerformPostLoad(Service* service) {
   shard_set->AwaitRunningOnShardQueue([](EngineShard* es) {
     es->search_indices()->RebuildAllIndices(
         OpArgs{es, nullptr, DbContext{&namespaces->GetDefaultNamespace(), 0, GetCurrentTimeMs()}});
+
+    // This clears fetched_items_, which may be set when loading an RDB file. We need to clear it
+    // to remove leftovers, see issue #4497
```

Comment:
while this is fine, I still think we should not bumpup during load.
can we also disable caching_mode_ in db_slice before we load and reenable afterwards?